__init__.py

```diff
@@ -92,51 +96,195 @@ def start_again(self):
+        open_shops = self.open_shops_search(shop_info)
+        if len(open_shops) >= 1:
+            return self.time_calculation(open_shops, True)
+        else:
+            return self.time_calculation(shop_info, False)
 
     def find_shop(self, user_request, mall_link):
+        """
+        When the intent is matched, user_request
+        variable contains the name of the shop.
+        The matching function get_shop_data() is
+        used to find the shop name in cache or
+        on the mall page.
+        If user's request is not None this function
+        can return several shops, one shop or empty
+        list.
+        If no shop was found asks user to repeat.
+        returns 1, user_request to continue the
+        execution loop in self.execute().
+        If there are several shops asks user what way
+        of sorting to choose: time, level, nothing.
+            If 'time' - finds open shops. If open shops
+            list is not empty speaks open shops, else
+            tells time difference between user and shops'
+            work hours.
+            If 'location' - asks what level user is interested
+            in. If shops were found speaks shops' info,
+            else tells that there is no shop on that level
+            and speaks all found shops.
+            If  'no' - sorts by time.
+            If nothing matched in the answer - sorts by time.
+        If there was one shop found speaks this
+        shop info. Returns 3, None to stop current
+        shop search.
+        Location and time sorting functions return
+        3, None to stop current shop search.
+        """
         LOG.info(str(user_request))
         LOG.info(str(mall_link))
         if user_request is not None:
             self.speak_dialog(f"I am parsing shops and malls for your request")
             LOG.info(f"I am parsing shops and malls for your request")
-            shop_info = self.request_handler.get_shop_data(mall_link, user_request, self.cache)
+            shop_info = self.request_handler.get_shop_data(mall_link, user_request)
+            LOG.info(f"I found {len(shop_info)} shops")
+            LOG.info(f"shop list: {shop_info}")
             if len(shop_info) == 0:
                 self.speak_dialog("shop_not_found")
                 user_request = self.get_response('repeat')
                 return 1, user_request
             elif len(shop_info) > 1:
-                return self.more_than_one(shop_info)
+                self.speak_dialog('more_than_one')
+                # ask for the way of selection: time, location, nothing
+                sorting_selection = self.get_response('Do you want to select'
+                                                      'store by work hours or location?')
+                if sorting_selection:
+                    LOG.info(f'Users answer on sorting options: {sorting_selection}')
+                    if 'time' in sorting_selection or 'hour' in sorting_selection:
```

Comment:
You could also pass a validator function to `get_response` or use `ask_selection` to simplify this logic. Consider using `self.voc_match` with `.voc` files in place of strings so this can be translated to other languages